### PR TITLE
Added missing tests to kernel32 and DbgHelp.

### LINKED ISSUE #224: Missing tests for MINIDUMP_TYPE, MiniDumpWriteDump, GetCurrentProcess, GetCurrentProcessId, GetCurrentThreadId.
There may be other apis, however there are missing tests that would check generation on all 3 of the TFMs the tests try to generate for.

If this gets tested it will expose possibly where the issue might be for #221 on these apis. Oddly enough it happens with the enum as well for some unknown reason that I think should generate the enum on .NET Standard 2.0 (but that is another issue for the one I linked) and yes I tried every version of CsWin32 and they all failed the same way on these.

### PR DESCRIPTION
**Pull request by AraHaan**
Fixes #224

(Tests created on my Mac)

### CODE REVIEW

**Comment by AraHaan** [comment]
Wait why doesn't the pipeline test ns2.0 as well so it can expose the failures on these when targeting that TFM?

**Comment by AArnott** [comment]
> why doesn't the pipeline test ns2.0 as well

netstandard2.0 isn't a runtime. It's just a standard. Tests can only run on a runtime like .NET Core, .NET Framework, or mono.
It looks like we only run tests on .NET 5, regardless of OS. For the code generation tests that seems adequate since I do not expect the runtime to influence the code generation itself. But we could add more runtimes if necessary.

**Comment by AArnott** [comment]
We don't want a test for every API to make sure it generated. We have tests for specific APIs that tend to be problematic and throw exceptions during generation, but ultimately those are redundant with our "generate everything" tests. Why do we need these new tests?

**Comment by AraHaan** [comment]
Because a lot of those api's I added are technically some of the common ones I find that people want to use on the internet, myself included.

**Comment by AraHaan** [comment]
I see this is why I have my issue on my codebase not generating this enum:
```
  Failed GeneratorTests.GetMINIDUMP_TYPEGeneration [126 ms]
  Error Message:
   Assert.True() Failure
Expected: True
Actual:   False
  Stack Trace:
     at GeneratorTests.GetMINIDUMP_TYPEGeneration() in D:\a\1\s\test\Microsoft.Windows.CsWin32.Tests\GeneratorTests.cs:line 248
```

Which is odd because it should be constant across all cpu specific builds of Windows 10 that has the latest dbghelp.dll.

**Comment by AArnott** [comment]
> Because a lot of those api's I added are technically some of the common ones I find that people want to use on the internet, myself included.

To be clear, I'm not arguing that we shouldn't generate those APIs. We do. I'm just saying they already have test coverage as part of the full-generation test. If we have a bug where certain APIs are simply *missing* in the generated code, then that's certainly worth taking a look at. It sounds like that's what you're saying, so I'll take another look.

File: test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs [hunk 1]
```diff
@@ -229,6 +232,33 @@ public void GetLastErrorNotIncludedInBulkGeneration()
...
+    /// Verifies that the MINIDUMP_TYPE enum is generated.
+    /// </summary>
+    // TODO: Figure out why this enum is not being generated and get it to generate as it should be constant with every platform.
+    [Fact]
+    public void GetMINIDUMP_TYPEGeneration()
+    {
+        this.generator = new Generator(this.metadataStream, DefaultTestGeneratorOptions, this.compilation, this.parseOptions);
+        Assert.True(this.generator.TryGenerate("MINIDUMP_TYPE", CancellationToken.None));
+        this.CollectGeneratedCode(this.generator);
+        Assert.True(this.generator.TryGetEnumName("MINIDUMP_TYPE", out string? actualDeclaringEnum));
```
Comment (AArnott) [inline test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs:245]: You're using the wrong API here. `TryGetEnumName` can't be used to verify that the enum is generated. 
By adding `this.AssertNoDiagnostics();` to the test the generated code is logged as test output and I see that the enum is in fact generated. So no bug here, it seems.

File: test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs [hunk 2]
```diff
@@ -229,6 +232,33 @@ public void GetLastErrorNotIncludedInBulkGeneration()
...
+        Assert.Equal("MINIDUMP_TYPE", actualDeclaringEnum);
+    }
+
+    /// <summary>
+    ///  Verifies that MiniDumpWriteDump is generated.
+    /// </summary>
+    // TODO: Make test try to generate using x86, x64, arm, and arm64 instead of AnyCPU so that it passes.
```
Comment (AArnott) [inline test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs:252]: The trick here is to add this line to the top of your test:

```cs
        this.compilation = this.compilation.WithOptions(this.compilation.Options.WithPlatform(Platform.X86));
```

With that, the test passes. So again, apparently no bug here.

File: test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs [hunk 3]
```diff
@@ -200,7 +200,10 @@ public void COMInterfaceWithSupportedOSPlatform(bool net50, bool allowMarshaling
             "ISpellCheckerFactory", // COM interface that includes `ref` parameters
             "LocalSystemTimeToLocalFileTime", // small step
             "ID3D12Resource", // COM interface with base types
-            "ID2D1RectangleGeometry")] // COM interface with base types
+            "ID2D1RectangleGeometry", // COM interface with base types
+            "GetCurrentProcess", // A handle for the current process
+            "GetCurrentProcessId", // An ID of the current process
+            "GetCurrentThreadId")] // An ID of the current thread of the current process
```
Comment (AArnott) [inline test/Microsoft.Windows.CsWin32.Tests/GeneratorTests.cs:206]: I don't see the value of testing these individually unless they provide a faster path to exposing an existing test failure in the `FullGeneration` test.

**Pull request closed without merging**